eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -1564,6 +1574,88 @@
     double sumOfDouble(DoubleFunction<? super T> function);
 
     /**
+     * @since 8.0
+     */
+    default IntSummaryStatistics summarizeInt(IntFunction<? super T> function)
+    {
+        IntSummaryStatistics stats = new IntSummaryStatistics();
+        this.each(each -> stats.accept(function.intValueOf(each)));
+        return stats;
+    }
+
+    /**
+     * @since 8.0
+     */
+    default DoubleSummaryStatistics summarizeFloat(FloatFunction<? super T> function)
+    {
+        DoubleSummaryStatistics stats = new DoubleSummaryStatistics();
+        this.each(each -> stats.accept(function.floatValueOf(each)));
+        return stats;
+    }
+
+    /**
+     * @since 8.0
+     */
+    default LongSummaryStatistics summarizeLong(LongFunction<? super T> function)
+    {
+        LongSummaryStatistics stats = new LongSummaryStatistics();
+        this.each(each -> stats.accept(function.longValueOf(each)));
+        return stats;
+    }
+
+    /**
+     * @since 8.0
+     */
+    default DoubleSummaryStatistics summarizeDouble(DoubleFunction<? super T> function)
+    {
+        DoubleSummaryStatistics stats = new DoubleSummaryStatistics();
+        this.each(each -> stats.accept(function.doubleValueOf(each)));
+        return stats;
+    }
+
+    /**
+     * This method produces the equivalent result as {@link Stream#collect(Collector)}.
+     *
+     * @since 8.0
+     */
+    default <R, A> R reduceInPlace(Collector<? super T, A, R> collector)
+    {
+        A mutableResult = collector.supplier().get();
+        BiConsumer<A, ? super T> accumulator = collector.accumulator();
+        this.each(each -> accumulator.accept(mutableResult, each));
+        return collector.finisher().apply(mutableResult);
+    }
+
+    /**
+     * This method produces the equivalent result as {@link Stream#collect(Supplier, BiConsumer, BiConsumer)}.
+     * The combiner used in collect is unnecessary in the serial case, so is not included in the API.
+     *
+     * @since 8.0
+     */
+    default <R> R reduceInPlace(Supplier<R> supplier, BiConsumer<R, ? super T> accumulator)
+    {
+        R result = supplier.get();
+        this.each(each ->  accumulator.accept(result, each));
+        return result;
+    }
+
+    /**
+     * This method produces the equivalent result as {@link Stream#reduce(BinaryOperator)}.
+     *
+     * @since 8.0
+     */
+    default Optional<T> reduce(BinaryOperator<T> accumulator)
+    {
+        if (this.isEmpty())
+        {
+            return Optional.empty();
+        }
+        T result = this.injectInto(null, (result1, each) ->
+                result1 == null ? each : accumulator.apply(result1, each));
+        return Optional.of(result);
+    }
```

Comment:
The method `isEmpty()` can be O(n) on `LazyIterable`s so I think that `reduce()` should be rewritten to only call `each()` or should be left abstract and implemented in subclasses.